pkg/templating/templating.go

```diff
@@ -47,10 +47,7 @@ func extractVariablesFromString(s, prefix string) ([]string, bool) {
 	vars := make([]string, len(matches))
 	for i, match := range matches {
 		groups := matchGroups(match, re)
-		// foo -> foo
-		// foo.bar -> foo
-		// foo.bar.baz -> foo
-		vars[i] = strings.SplitN(groups["var"], ".", 2)[0]
+		vars[i] = groups["var"]
```

Comment:
I'm having a hard time figuring out if the logic has remained the same here or has been changed to make an intentional difference to behaviour. Would you mind just quickly describing what this change is?

Reply:
Yes, I intentionally made this change. 

I added a new test to the templating package which hopefully provides additional insight as to why this was needed. This new test would fail without this change. 

Basically, `strings.SplitN(groups["var"], ".", 2)[0]` was just returning the first split of a variable. So if groups["var"] was set to `foo.bar.baz`, it would just return `foo`.

Since the `vars` map passed into `ValidateVariable` can contain nested variables now, like `fooImage.url`, I believe we want to return the entire variable string to validate against.

Reply:
OK got it, thankyou, I understand the original issue and this fix more clearly now!